src/Plugins.js

```diff
@@ -0,0 +1,168 @@
+        debug(`Botium plugin loaded from ${filename}, but its name ${plugin.PluginDesc.name} is invalid, will be overwritten with ${pluginName}`)
+      }
+
+      const PluginDesc = plugin.PluginDesc || {}
+      return {
+        PluginVersion: plugin.PluginVersion || '1',
+        PluginType,
+        PluginDesc: Object.assign(
+          { description: pluginName },
+          PluginDesc,
+          {
+            name: pluginName,
+            capabilities: PluginDesc.capabilities ? JSON.stringify(PluginDesc.capabilities) : null
+          }
+        )
+      }
+    } catch (err) {
+      debug(`Loading Botium plugin from ${filename} failed - ${err.message}`)
+      return null
+    }
+  }
+  return null
+}
+
+const getOtherPlugin = (filename, pathToRes, type) => {
+  let PluginType = null
+  if (type !== PLUGIN_TYPE_CONNECTOR && filename.toLowerCase().startsWith(TYPE_TO_PREFIX[type])) {
+    PluginType = type
+  }
+  if (!PluginType) {
+    return null
+  }
+  try {
+    const plugin = require(filename)
+    const pluginPath = require.resolve(filename)
+    if (!pluginPath.startsWith(pathToRes)) {
+      debug(`Plugin mismatch error. Plugin ${path.resolve(pathToRes, filename)} loaded from ${pluginPath}! `)
+      return null
+    }
+    if (plugin.PluginType && plugin.PluginType !== PluginType) {
+      debug(`Botium plugin loaded from ${filename}, but its type ${plugin.PluginType} is invalid, will be overwritten with ${PluginType}`)
+    }
+
+    let pluginName = filename.substring(TYPE_TO_PREFIX[type].length)
+    if (pluginName.toLowerCase().endsWith('.js')) {
+      pluginName = pluginName.substring(0, pluginName.length - '.js'.length)
+    }
+    if (plugin.PluginDesc && plugin.PluginDesc.name && plugin.PluginDesc.name !== pluginName) {
+      debug(`Botium plugin loaded from ${filename}, but its name ${plugin.PluginDesc.name} is invalid, will be overwritten with ${pluginName}`)
+    }
+
+    const PluginDesc = plugin.PluginDesc || {}
+
+    return {
+      PluginVersion: plugin.PluginVersion || '1',
+      PluginType,
+      PluginDesc: {
+        name: pluginName,
+        description: PluginDesc.description,
+        type: TYPE_TO_COMPONENT_TYPE[type],
+        src: filename,
+        ref: PluginDesc.ref || pluginName.toUpperCase(),
+        global: PluginDesc.global || false,
+        args: PluginDesc.args || '{}'
+      }
+    }
+  } catch (err) {
+    debug(`Loading Botium plugin from ${filename} failed - ${err.message}`)
+    return null
+  }
+}
+
+const TYPE_TO_FN = {
+  PLUGIN_TYPE_CONNECTOR: getConnectorPlugin,
+  PLUGIN_TYPE_ASSERTER: getOtherPlugin,
+  PLUGIN_TYPE_LOGICHOOK: getOtherPlugin,
+  PLUGIN_TYPE_USERINPUT: getOtherPlugin
+}
+
+const getPlugins = (type) => {
+  if (!TYPE_TO_FN[type]) {
+    debug(`Invalid plugin type "${type}"`)
+    return Promise.resolve([])
+  }
+  return new Promise((resolve, reject) => {
+    const pathToRes = path.resolve(process.cwd(), 'resources')
```

Comment:
Please make this path configurable, with the default fallback of the local "resources" folder.